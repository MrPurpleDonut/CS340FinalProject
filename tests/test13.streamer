funk num sum(list<num> nums)
{
  num result = 0;
  for (num n : nums)
  {
    result = result + n;
  }
  return result;
}

