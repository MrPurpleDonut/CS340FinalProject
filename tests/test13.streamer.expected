Program parsed correctly!
---------------------------
--------Printing AST-------
---------------------------
Program
    Function List
        Function declaration: sum
            Type: num
            Parameter List
                Parameter: nums
                    Type: list
                        Type: num
            Statement List
                Declaration statement: result
                    Type: num
                    Assignment statement: result
                        Number: 0.0
                For Statement: n
                    Type: num
                    Identifier: nums
                    Statement List
                        Assignment statement: result
                            Binary operator '+'
                                Identifier: result
                                Identifier: n
                Return Statement
                    Identifier: result
    Statement List
